integration/cosmos.md

```diff
@@ -0,0 +1,163 @@
+# Keystone Cosmos Mode Integration Guide
+
+The goal of the Keystone hardware wallet is to provide users with a reliable and quality signer by protecting users'
+private keys so Keystone seeks to integrate with existing wallets and applications.
+
+
+## Firmware Requirement
+
+You need Keystone Multi-coin Firmware support Cosmos for the integration. Find it
+from [here](https://keyst.one/firmware?locale=en).
+
+## Wallet Integration
+
+In order to work with Keystone, a wallet can follow the process:
+
+1. Keystone provides the public key information to the wallet via QR code, to generate Cosmos addresses.
+2. The wallet generates the unsigned transactions, sends it to Keystone to sign via QR Code.
+3. Keystone would then sign the data and provide a signature as well as the related public key, back to the wallet via QR Code.
+4. The wallet receives the signature and public key, constructs the signed data (transaction), and performs the following activities like broadcasting the transaction, etc.
+
+## Library
+
+[`@keystonehq/bc-ur-registry-cosmos`](https://www.npmjs.com/package/@keystonehq/bc-ur-registry-cosmos)
+
+[`@keystonehq/animated-qr`](https://www.npmjs.com/package/@keystonehq/animated-qr)
+
+[`@keystonehq/cosmos-keyring`](https://www.npmjs.com/package/@keystonehq/cosmos-keyring)
+
+## Example
+
+With the help of `@keystone/cosmos-keyring`, wallets can easily integrate with Keystone without caring too much about UR.
+`@keystone/cosmos-keyring` encapsulates the details like encode/decode UR, construct UR from data etc. 
+
+Wallets only need to pay some attention to the UI like display the qr code or render the camera to scan the QR
+code.
+
+`@keystone/cosmos-keyring` uses `InteactionProvider` to handle the UI interaction, by default it uses `@keystone/sdk` as
+the interaction provider to interact with wallet, it's functional but less UI designed, you can define your own
+interaction provider by extending the `BaseKeyring`. With `DefaultKeyring` the integration is very simple:
+
+### Connect Keystone
+
+```ts
+import {DefaultKeyring} from '@keystonehq/cosmos-keyring';
+
+const connectWallet = aysnc()=>{
+...
+    // Init a keyring instance 
+    const keyring = DefaultKeyring.getEmptyKeyring();
+
+    // This will call the video camera.
+    // by scanning the QR code displayed in Keystone HardWare Wallet, The keyring will be initalized with public key, hdPath inforamtion.
+    await keyring.readKeyring();
+   console.log(keyring.getPubKeys()[0].pubKey);  // 8e53e7b10656816de70824e3016fc1a277e77825e12825dc4f239f418ab2e04e
+...
+}
+
+```
+
+### Sign Transaction
+
+```ts
+// In your Transfer component, call the `keyring.signDirectTransaction` or 'keyring.signAminoTransaction', up to your sign mdoe.
+// It will render the `cosmos-sign-request` UR. 
+// This promised will be resolved by scanning the `cosmos-signature` UR generated from Keystone HardWare Wallet.
+const {signature, pubKey} = await keyring.signDirectTransaction(pubKey, signData);
```

Comment:
construct sign data  example for `signDirectTransaction` and `signAminoTransaction `